### PR TITLE
[POC] Export calculation functions in order to allow unit tests

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Our development workflow doesn't allow us to create unit tests (for the product page v2 for example), we could avoid this by exporting functional things into functions (using functional programming instead of class in order to simplify the data flow understanding)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Product page price calculation should work as expected
| Possible impacts? | Product page v2

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25909)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  let newTaxRate;
+
+  try {
+    newTaxRate = new BigNumber(taxRate);
+  } catch (error) {
+    newTaxRate = new BigNumber(NaN);
+  }
+
+  if (newTaxRate.isNaN()) {
+    newTaxRate = new BigNumber(0);
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts]: Normally you would throw an exception here

File: admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import BigNumber from 'bignumber.js';
+
+export const calculateTax = (taxRate: string): BigNumber => {
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts]: This doesn't calculate any tax, this takes a tax rate and outputs a tax multiplier.
This is why it's important to describe functions in doc headers.

File: admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts]: I'm worried that we might end up with hundreds of "helpers" scattered all over

File: admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts (hunk 2 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts]: It's not even about tax, it's just a multiplier, I fixed this

File: admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts (hunk 1 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts]: The exception is already thrown by BigNumber tbh and the logic should probably be handler by the parent component if the tax ratio is undefined don't you think?

**Pull request closed without merging**